deploy/buildtools/buildfarm.py

```diff
@@ -0,0 +1,334 @@
+import logging
+import sys
+import abc
+
+from awstools.awstools import *
+
+# imports needed for python type checking
+from typing import cast, Any, Dict, Optional, Sequence, List, TYPE_CHECKING
+from mypy_boto3_ec2.service_resource import Instance as EC2InstanceResource
+# needed to avoid type-hint circular dependencies
+# TODO: Solved in 3.7.+ by "from __future__ import annotations" (see https://stackoverflow.com/questions/33837918/type-hints-solve-circular-dependency)
+#       and normal "import <module> as ..." syntax (see https://www.reddit.com/r/Python/comments/cug90e/how_to_not_create_circular_dependencies_when/)
+if TYPE_CHECKING:
+    from buildtools.buildconfig import BuildConfig
+else:
+    BuildConfig = object
+
+rootLogger = logging.getLogger()
+
+class BuildHost:
+    """Abstract class representing a single build host which holds a single build config.
```

Comment:
No, you mean this is a Basic Build Host that can be expanded with extra functionality cuz that's what you're doing with EC2BuildHost.

Reply:
The comment here needs to be reworded a bit. This isn't an abstract class, this can be used as a build host (hence why it doesn't inherit from `abc`). I'll update the wording.